code/mission/import/xwingmissionparse.cpp

```diff
@@ -623,6 +624,245 @@ void parse_xwi_flightgroup(mission *pm, const XWingMission *xwim, const XWMFligh
+	case XWMObjectFormation::ojf_SideYZ:
+		return vm_vec_new(objectPosX, (objectPosZ + offsetAxisA), (objectPosY + offsetAxisB));
+	case XWMObjectFormation::ojf_FrontXZ:
+		return vm_vec_new((objectPosX + offsetAxisA), (objectPosZ + offsetAxisB), objectPosY);
+	case XWMObjectFormation::ojf_Scattered:
+		return vm_vec_new(objectPosX, objectPosZ, objectPosY);
+	default:
+		break;
+	}
+	return vm_vec_new(objectPosX, objectPosZ, objectPosY);
+}
+
+void xwi_determine_object_orient(matrix* orient, const XWMObject* oj)
+{
+	angles a;
+	a.p = oj->object_pitch;
+	a.b = oj->object_roll;
+	a.h = oj->object_yaw;
+	vm_angles_2_matrix(orient, &a);
+	return;
+}
+
+// Determine the unique name from the object comprised of the object type and suffix. 
+// Add the new name to the objectNameSet and return it to parse_xwi_objectgroup
+const char* xwi_determine_space_object_name(SCP_set<SCP_string>& objectNameSet, const char* class_name)
+{
+	char base_name[NAME_LENGTH];
+	char suffix[NAME_LENGTH];
+	strcpy_s(base_name, class_name);
+	end_string_at_first_hash_symbol(base_name);
+
+	// we'll need to try suffixes starting at 1 and going until we find a unique name
+	int n = 1;
+	char object_name[NAME_LENGTH];
+	do {
+		sprintf(suffix, NOX(" %d"), n++);
+
+		// start building name
+		strcpy_s(object_name, base_name);
+
+		// if generated name will be longer than allowable name, truncate the class section of the name by the overflow
+		int char_overflow = static_cast<int>(strlen(base_name) + strlen(suffix)) - (NAME_LENGTH - 1);
+		if (char_overflow > 0) {
+			object_name[strlen(base_name) - static_cast<size_t>(char_overflow)] = '\0';
+		}
+
+		// complete building the name by adding suffix number and converting case
+		strcat_s(object_name, suffix);
+		SCP_totitle(object_name);
+
+		// continue as long as we find the name in our set
+	} while (objectNameSet.find(object_name) != objectNameSet.end());
+
+	// name does not yet exist in the set, so it's valid; add it and return
+	auto iter = objectNameSet.insert(object_name);
+	return iter.first->c_str();
+}
+
+void parse_xwi_objectgroup(mission* pm, const XWingMission* xwim, const XWMObject* oj)
+{
+	SCP_UNUSED(pm);
+	auto class_name = xwi_determine_object_class(oj);
+	if (class_name == nullptr)
+		return;
+
+	int number_of_objects = oj->numberOfObjects;
+	if (number_of_objects < 1)
+		return;
+
+	// object position and orientation
+	// NOTE: Y and Z are swapped after all operartions are perfomed
+    // units are in kilometers (after processing by xwinglib which handles the factor of 160), so scale them up
+	float objectPosX = oj->object_x*1000;
+	float objectPosY = oj->object_y*1000;
+	float objectPosZ = oj->object_z*1000;
+	float offsetAxisA = 0;
+	float offsetAxisB = 0;
+
+	int mine_dist = 400; // change this to change the distance between the mines
+	int mine_laser_index = weapon_info_lookup("T&B KX-5#imp"); // "Defense Mine#Ion" needs to have its weapon changed to laser
```

Comment:
As long as you're here, add this:
```
auto weapon_name = "T&B KX-5#imp";
```
Then use `weapon_name` in both the lookup and the warning.  See how `team_name` is handled below.